src/java.base/share/classes/java/lang/foreign/FunctionDescriptor.java

```diff
@@ -94,6 +95,22 @@ public sealed interface FunctionDescriptor permits FunctionDescriptorImpl {
      */
     FunctionDescriptor dropReturnLayout();
 
+    /**
+     * Returns the method type consisting of the carrier types of the layouts in this function descriptor.
+     * <p>
+     * The carrier type of a layout is determined as follows:
+     * <ul>
+     * <li>If the layout is a {@link ValueLayout} the carrier type is determined through {@link ValueLayout#carrier()}.</li>
+     * <li>If the layout is a {@link GroupLayout} the carrier type is {@link MemorySegment}.</li>
+     * <li>If the layout is a {@link PaddingLayout}, or {@link SequenceLayout} an {@link IllegalArgumentException} is thrown.</li>
+     * </ul>
+     *
+     * @return the method type consisting of the carrier types of the layouts in this function descriptor
+     * @throws IllegalArgumentException if one or more layouts in the function descriptor can not be mapped to carrier
+     *                                  types (e.g. if they are sequence layouts or padding layouts).
```

Comment:
```suggestion
     *                                  types (e.g. if they are padding layouts).
```

Reply:
I've removed the support for SequenceLayout now, so I think this is currently correct?

Reply:
Ah - of course. I didn't notice the bullet being rewritten. I think this is a good move, and one that still give us the option to change things later, if needed (it's always relatively ok to turn throwing cases into non-throwing ones).